Intro to computer networking (#16)
Kalschatzi/learning_platform: content/module1/_index.md

weight = 1
+++

## Java and OOP introduction

### Module Objectives

This is an introductory module to start understanding the key concepts of [Object Oriented Programming - OOP](https://www.geeksforgeeks.org/introduction-of-object-oriented-programming/).

> We'll be using Java, as it's widely use, a run-everywhere language thanks to the [JVM](https://www.geeksforgeeks.org/jvm-works-jvm-architecture/) portability. It's a somewhat verbose language which has positives and negatives - main positive as far as we're concerned is that it's a great introductory OOP language, very widely used, with loads of resources ready to use.

Reviewer: ```suggestion
We'll use Java, as it's widely used, a run-everywhere language thanks to the [JVM](https://www.geeksforgeeks.org/jvm-works-jvm-architecture/) portability. It's a somewhat verbose language which has positives and negatives - main positive is that it's a great introductory OOP language, very widely used, with loads of resources ready to use.
```

It will then cover basic algorithms and data structures. 

> We'll understand how to evaluate the complexities of an algorithms We'll look into specific data structures, their advantages, disadvantages, when you should use them and why.

Reviewer: ```suggestion
We'll understand how to evaluate the complexities of algorithms. We'll look into specific data structures, their advantages, disadvantages and use cases.
```
